test/metabase/api/search_test.clj

```diff
@@ -150,42 +150,46 @@
-                                :as action}   (merge (data-map "action %s action")
-                                                     {:type :query, :model_id (u/the-id action-model)})
-                   Database    {db-id :id
-                                :as db}       (data-map "database %s database")
-                   Table       table          (merge (data-map "database %s database")
-                                                     {:db_id db-id})
-
-                   QueryAction _qa (query-action action-id)
-                   Card        card           (coll-data-map "card %s card" coll)
-                   Card        dataset        (assoc (coll-data-map "dataset %s dataset" coll)
-                                                     :type :model)
-                   Dashboard   dashboard      (coll-data-map "dashboard %s dashboard" coll)
-                   Card        metric         (assoc (coll-data-map "metric %s metric" coll)
-                                                     :type :metric)
-                   Segment     segment        (data-map "segment %s segment")]
-      (f {:action     action
-          :collection coll
-          :card       card
-          :database   db
-          :dataset    dataset
-          :dashboard  dashboard
-          :metric     metric
-          :table      table
-          :segment    segment}))))
+    (search.tu/with-temp-index-table
+      (mt/with-temp [Collection  coll           (data-map "collection %s collection")
+                     Card        action-model   (if in-root-collection?
+                                                  action-model-params
+                                                  (assoc action-model-params :collection_id (u/the-id coll)))
+                     Action      {action-id :id
+                                  :as action}   (merge (data-map "action %s action")
+                                                       {:type :query, :model_id (u/the-id action-model)})
+                     Database    {db-id :id
+                                  :as db}       (data-map "database %s database")
+                     Table       table          (merge (data-map "database %s database")
+                                                       {:db_id db-id})
+
+                     QueryAction _qa (query-action action-id)
+                     Card        card           (coll-data-map "card %s card" coll)
+                     Card        dataset        (assoc (coll-data-map "dataset %s dataset" coll)
+                                                       :type :model)
+                     Dashboard   dashboard      (coll-data-map "dashboard %s dashboard" coll)
+                     Card        metric         (assoc (coll-data-map "metric %s metric" coll)
+                                                       :type :metric)
+                     Segment     segment        (data-map "segment %s segment")]
+        (search/reindex!)
```

Comment:
Normally the hooks would take care of it, but in dev they are async so I added a synchronous call as a stop gap.

Reply:
Found a better fix - switch to synchronous when we switch to working on a thread-local index. The worker would write to the wrong index in any case.